[CI] S3BlobStoreRepositoryMetricsTests testMetricsForRequestRangeNotSatisfied failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #3628 / windows-2016_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/lj7uzex4rmuvw)
- [elasticsearch-periodic-platform-support #3628 / windows-2019_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/iun3exwf4dhdo)
- [elasticsearch-periodic-platform-support #3628 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/7ofayn6c6jmug)
- [elasticsearch-periodic-platform-support #3628 / amazonlinux-2_platform-support-aws](https://gradle-enterprise.elastic.co/s/4azbvnts3xlbm)
- [elasticsearch-periodic-platform-support #3624 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/j63bcpjzua62o)
- [elasticsearch-periodic-platform-support #3624 / almalinux-8-aarch64_checkpart1_platform-support-arm](https://gradle-enterprise.elastic.co/s/zofpzjoj3cohs)
- [elasticsearch-periodic #3650 / release-tests](https://gradle-enterprise.elastic.co/s/k3xrw5oi4wngi)
- [elasticsearch-periodic-platform-support #3624 / ubuntu-2004-aarch64_checkpart1_platform-support-arm](https://gradle-enterprise.elastic.co/s/zyxtrd724ux6m)
- [elasticsearch-periodic-platform-support #3624 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/ymh64mjmuwcom)
- [elasticsearch-periodic-platform-support #3624 / opensuse-leap-15_platform-support-unix](https://gradle-enterprise.elastic.co/s/6uawjpqym4mjw)

**Reproduction Line:**
```
gradlew ":modules:repository-s3:internalClusterTest" --tests "org.elasticsearch.repositories.s3.S3BlobStoreRepositoryMetricsTests.testMetricsForRequestRangeNotSatisfied" -Dtests.seed=32B63F05405BDCDB -Dtests.locale=pt -Dtests.timezone=Asia/Kashgar -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testMetricsForRequestRangeNotSatisfied'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.repositories.s3.S3BlobStoreRepositoryMetricsTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <2L>
     but: was <1L>
```

**Issue Reasons:**
- [main] 3 consecutive failures in test testMetricsForRequestRangeNotSatisfied
- [main] 3 consecutive failures in step windows-2022_checkpart1_platform-support-windows
- [main] 3 consecutive failures in step amazonlinux-2_platform-support-aws
- [main] 22 failures in test testMetricsForRequestRangeNotSatisfied (15.9% fail rate in 138 executions)
- [main] 3 failures in step windows-2016_checkpart1_platform-support-windows (75.0% fail rate in 4 executions)
- [main] 3 failures in step windows-2022_checkpart1_platform-support-windows (75.0% fail rate in 4 executions)
- [main] 2 failures in step ubuntu-2004-aarch64_checkpart1_platform-support-arm (50.0% fail rate in 4 executions)
- [main] 3 failures in step amazonlinux-2_platform-support-aws (100.0% fail rate in 3 executions)
- [main] 4 failures in pipeline elasticsearch-periodic-platform-support (100.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java
+++ b/modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java
@@ -224,11 +224,13 @@ class S3BlobStore implements BlobStore {
                 return;
             }
 
-            final long totalTimeInMillis = getTotalTimeInMillis(requestTimesIncludingRetries);
-            if (totalTimeInMillis == 0) {
+            final long totalTimeInNanos = getTotalTimeInNanos(requestTimesIncludingRetries);
+            if (totalTimeInNanos == 0) {
                 logger.warn("Expected HttpRequestTime to be tracked for request [{}] but found no count.", request);
             } else {
-                s3RepositoriesMetrics.common().httpRequestTimeInMillisHistogram().record(totalTimeInMillis, attributes);
+                s3RepositoriesMetrics.common()
+                    .httpRequestTimeInMillisHistogram()
+                    .record(TimeUnit.NANOSECONDS.toMillis(totalTimeInNanos), attributes);
             }
         }
 
@@ -271,19 +273,20 @@ class S3BlobStore implements BlobStore {
         }
     }
 
-    private static long getTotalTimeInMillis(List<TimingInfo> requestTimesIncludingRetries) {
-        // Here we calculate the timing in Milliseconds for the sum of the individual subMeasurements with the goal of deriving the TTFB
-        // (time to first byte). We calculate the time in millis for later use with an APM style counter (exposed as a long), rather than
-        // using the default double exposed by getTimeTakenMillisIfKnown(). We don't need sub-millisecond precision. So no need perform
-        // the data type castings.
-        long totalTimeInMillis = 0;
+    private static long getTotalTimeInNanos(List<TimingInfo> requestTimesIncludingRetries) {
+        // Here we calculate the timing in Nanoseconds for the sum of the individual subMeasurements with the goal of deriving the TTFB
+        // (time to first byte). We use high precision time here to tell from the case when request time metric is missing (0).
+        // The time is converted to milliseconds for later use with an APM style counter (exposed as a long), rather than using the
+        // default double exposed by getTimeTakenMillisIfKnown().
+        // We don't need sub-millisecond precision. So no need perform the data type castings.
+        long totalTimeInNanos = 0;
         for (TimingInfo timingInfo : requestTimesIncludingRetries) {
             var endTimeInNanos = timingInfo.getEndTimeNanoIfKnown();
             if (endTimeInNanos != null) {
-                totalTimeInMillis += TimeUnit.NANOSECONDS.toMillis(endTimeInNanos - timingInfo.getStartTimeNano());
+                totalTimeInNanos += endTimeInNanos - timingInfo.getStartTimeNano();
             }
         }
-        return totalTimeInMillis;
+        return totalTimeInNanos;
     }
 
     @Override

